### DISCUSSION TITLE
Suricata bpf

### DISCUSSION THREAD

**Discussion by klipinski110** [Security-Onion-Solutions/securityonion#7887] [Unsupported Versions]
Good morning, with more bpf rules in suricata pops up error unknown argument. The length of the bpf file when an error occurs is approximately 1100 characters. Is there any solution? Regards

**Comment by UMHB-InfoSec** [comment]
_Please read the following with the knowledge that I am not attempting to insult or belittle in any way._

You may want to read this:

[Read before posting! #1720](https://github.com/Security-Onion-Solutions/securityonion/discussions/1720)

Particularly the last section

```
Provide sufficient technical info
In order to be as effective and efficient as possible, please consider the following when posing your question/problem to the group:
http://www.chiark.greenend.org.uk/~sgtatham/bugs.html

Please include the following details where you can:

Security Onion version as seen in the lower left corner of SOC and in /etc/soversion. For example: 2.3.110
Is this a cloud deployment or on-prem? If on-prem, do you have Internet access or this an airgap installation?
Did you install from our Security Onion ISO image or did you perform a network installation?
If network installation, did you install on CentOS 7 or Ubuntu?
How many nodes do you have?
What are the hardware specs of each of those nodes?
How are each of those nodes configured? (ex. manager with 2 search nodes and 3 forward nodes)
Are you experiencing issues monitoring network traffic? If so, are you sniffing from a tap or span port and what is the traffic volume?
Does so-status show all services running?
Do you get any failures when you run sudo salt-call state.highstate?
Does the SOC Grid page show any failures?
Explain your issue. For example: Installation fails when I select this series of options...
Provide applicable logs. If you are having problems right after setup, provide /root/sosetup.log. If you are having problems during soup, provide /root/soup.log. If you are having problems with a specific component, provide that component's logs from /opt/so/log/.
If you need to attach files, please make sure they are plain text format. No Word docs or PDFs please.
```

We want to help but we need info in order to help you.  Most of the people here have other full-time responsibilities and we just don't have the time to pull the info out of you.  The SO Solutions people who admin this repo are providing their time and energy for free and we need to be respectful of that.

**Comment by klipinski110** [comment]
I'm using ubuntu 20.04 with distributed deployment.Security onion version 
Sensor configuration file:host:
  mainint: 'eth0'
sensoroni:
  node_address: '192.168.15.50'
  node_description: ''

sensor:
  interface: 'bond0'
  mainip: '192.168.15.50'
  mainint: 'eth0'
  zeek_lbprocs: 3
  suriprocs: 12
  manager: 'onionlan'
  mtu: 1500
  uniqueid: 1650884684
steno:
  enabled: True
patch:
  os:
    source: 'direct'
    schedule_name: 'auto'
    enabled: True
    splay: 300

elasticsearch:
  mainip: '192.168.15.50'
  mainint: 'eth0'
  esheap: '10950m'
  esclustername: '{{ grains.host }}'
  node_type: 'search'
  es_port: 9200
  log_size_limit: 68
  node_route_type: 'hot'

logstash_settings:
  ls_pipeline_batch_size: 125
  ls_input_threads: 1
  lsheap: 1000m
  ls_pipeline_workers: 16

suricata:
    config:
          max-pending-packets: 10024
          default-packet-size: 2515
          nflog:
            buffer-size: 28432
            max-size: 30000
          pcre:
            match-limit: 4500
            match-limit-recursion: 2500
          defrag:
               memcap: 512mb
               hash-size: 75536
          flow:
               memcap: 1gb
               hash-size: 75536
          stream:
               memcap: 512mb
               reassembly:
                        memcap: 2gb
                        depth:  2mb
          host:
               memcap: 512mb
          af-packet:
             - interface: bond0
               cluster-id: 59
               cluster-type: cluster_flow
               defrag: true
               use-mmap: true
               threads: 12
               tpacket-v3: true
               ring-size: 500000
             - interface: default
               
nids:
  bpf:         
        - (not (host 192.168.11.5 or host 192.168.11.11 or host 192.168.11.88 or host 192.168.11.95 or host 192.168.11.134 or host 192.168.11.209 and proto ICMP)) &&
        - (not (host 192.168.7.14 or host 192.168.7.45 or host 192.168.7.46 or host 192.168.7.56 or host 192.168.7.57 or 192.168.7.58 or host 192.168.7.68 or host 192.168.7.69 or host 192.168.7.85 or host 192.168.7.87 or host 192.168.7.91 or host 192.168.7.92 or host 192.
168.7.93 or host 192.168.7.125 or host 192.168.7.126 or host 192.168.70.19 or host 192.168.70.20 or host 192.168.70.24 or host 192.168.70.26 or host 192.168.70.29 or host 192.168.70.31 or host 192.168.70.32 or host 192.168.70.33 or host 192.168.70.34 or host 192.168.70.35
 or host 192.168.70.38 or host 192.168.70.39 or host 192.168.70.43 or host 192.168.70.45 or host 192.168.70.50 or host 192.168.70.53 or host 192.168.70.55 or host 192.168.70.56 or host 192.168.70.61 or host 192.168.70.63 or host 192.168.70.64 or host 192.168.70.65 or host
 192.168.70.66 or host 192.168.70.67 or host 192.168.70.88 or host 192.168.70.91 or host 192.168.70.93 or host 192.168.70.95 or host 192.168.70.96 or host 192.168.70.97 or host 192.168.70.98 or host 192.168.77.55 or host 192.168.77.57 or host 192.168.77.101 or host 192.16
8.77.102 or host 192.168.77.110 or host 192.168.77.111 or host 192.168.77.113 or host 192.168.77.115 or host 192.168.77.140 and host 172.17.0.3 and port 53 or port 139 or port 445)) &&
        - (not (host 192.168.7.14 or host 192.168.7.33 or host 192.168.7.35 or host 192.168.7.45 or host 192.168.7.46 or host 192.168.7.69 or host 192.168.7.87 or host 192.168.7.91 or host 192.168.7.92 or host 192.168.70.19 or host 192.168.70.20 or host 192.168.70.24 or h
ost 192.168.70.35 or host 192.168.70.43 or host 192.168.70.26 or host 192.168.70.32 or host 192.168.70.33 or host 192.168.70.34 or host 192.168.70.35 or host 192.168.70.38 or host 192.168.70.45 or host 192.168.70.64 or host 192.168.70.67 or host 192.168.70.87 or host 192.
168.70.96 or host 192.168.77.102 and host 172.17.0.4 and port 445)) &&
        - (not (host 192.168.7.14 or host 192.168.7.68 or host 192.168.7.91 or host 192.168.7.92 or host 192.168.70.19 or host 192.168.70.26 or host 192.168.70.32 or host 192.168.70.33 or host 192.168.70.35 or host 192.168.70.38 or host 192.168.70.39 or host 192.168.70.43
 or host 192.168.70.45 or host 192.168.70.50 or host 192.168.70.55 or host 192.168.70.64 or host 192.168.70.66 or host 192.168.70.67 or host 192.168.70.96 or host 192.168.77.102 and host 172.17.0.4 and port 57905)) &&
        - (not (host 192.168.7.45 or host 192.168.7.46 or host 192.168.7.68 or host 192.168.7.69 or host 192.168.7.87 or host 192.168.7.91 or host 192.168.7.92 or host 192.168.7.93 or host 192.168.70.19 or host 192.168.70.24 or host 192.168.70.26 or host 192.168.70.32 or
host 192.168.70.33 or host 192.168.70.34 or host 192.168.70.35 or host 192.168.70.38 or host 192.168.70.39 or host 192.168.70.45 or host 192.168.70.55 or host 192.168.70.66 or host 192.168.70.61 or host 192.168.70.64 or host 192.168.70.66 or host 192.168.70.67 or host 192
.168.70.96 and host 172.17.0.6 and port 445)) &&
        - (not (host 192.168.7.45 and host 192.168.7.1 and port 53)) &&
        - (not (host 192.168.70.20 or host 192.168.70.38 or host 192.168.70.43 or host 192.168.70.45 or host 192.168.70.67 and host 192.168.70.1 and port 53)) &&
        - (not (host 192.168.7.46 or host 192.168.7.199 or host 192.168.70.33 or host 192.168.70.35 and host 192.168.6.65 and port 445)) &&
        - (not (host 192.168.7.86 or host 192.168.7.175 or host 192.168.7.196 or host 192.168.7.199 or host 192.168.7.234 and host 192.168.6.65 and port 111)) &&
        - (not (host 192.168.7.84 and host 192.168.6.65 and port 139)) &&
        - (not (host 192.168.70.35 or host 192.168.70.38 or host 192.168.70.39 or host 192.168.70.66 or host 192.168.70.67 or host 192.168.70.96 and host 192.168.6.162 and port 445)) &&
        - (not (host 192.168.7.93 or host 192.168.70.20 or host 192.168.70.32 or host 192.168.70.33 or host 192.168.70.34 and host 192.168.7.207 and port 5432)) &&
        - (not (host 192.168.77.90 and host 192.168.6.121 and port 5432)) &&
        - (not (host 192.168.70.20 or host 192.168.70.56 and host 192.168.6.221 and port 5432)) &&
        - (not (host 192.168.7.14 or host 192.168.70.43 and host 192.168.4.84 and port 5432)) &&
        - (not (host 192.168.77.110 or host 192.168.77.111 and host 192.168.7.116 and port 80)) &&
        - (not (host 192.168.77.101 and host 192.168.7.117 and port 80)) &&
        - (not (host 192.168.7.45 or host 192.168.70.67 and host 192.168.7.134 and port 80)) &&
        - (not (host 192.168.88.25 or host 192.168.70.67 and host 192.168.7.134 and port 161)) &&
        - (not (host 192.168.70.39 and host 192.168.4.196 and port 8080)) &&
        - (not (host 192.168.70.33 or host 192.168.70.35 or host 192.168.70.96 and host 192.168.7.202 and port 161)) &&
        - (not (host 192.168.7.70 or host 192.168.7.235 or host 192.168.77.35 or host 192.168.77.36 or host 192.168.77.37 or host 192.168.77.39 or host 192.168.77.40 or host 192.168.77.41 or host 192.168.77.51 or host 192.168.77.55 or host 192.168.77.56 or host 192.168.77
.57 or host 192.168.77.58 or host 192.168.77.59 or host 192.168.77.61 or host 192.168.77.63 or host 192.168.77.65 or host 192.168.77.70 or host 192.168.77.71 or host 192.168.77.74 or host 192.168.77.75 or host 192.168.77.78 or host 192.168.77.84 or host 192.168.77.85 or h
ost 192.168.77.88 or host 192.168.77.89 or host 192.168.77.90 or host 192.168.77.91 or host 192.168.77.93 or host 192.168.77.96 or host 192.168.77.97 or host 192.168.77.98 or host 192.168.77.99 or host 192.168.77.137 or host 192.168.77.139 or host 192.168.77.140 and host
172.17.5.10 and port 21)) &&
        - (not (host 192.168.7.51 or host 192.168.7.70 or host 192.168.7.86 or host 192.168.7.94 or host 192.168.7.95 or host 192.168.7.109 or host 192.168.7.110 or host 192.168.7.112 or host 192.168.7.113 or host 192.168.7.116 or host 192.168.7.117 or host 192.168.7.118
or host 192.168.7.119 or host 192.168.7.136 or host 192.168.7.172 or host 192.168.7.173 or host 192.168.7.192 or host 192.168.7.196 or host 192.168.7.202 or host 192.168.7.206 or host 192.168.7.208 or host 192.168.7.209 or host 192.168.7.246 or host 192.168.70.98 or host
192.168.70.213 or host 192.168.70.225 or host 192.168.77.35 or host 192.168.77.37 or host 192.168.77.39 or host 192.168.77.40 or host 192.168.77.41 or host 192.168.77.51 or host 192.168.77.55 or host 192.168.77.56 or host 192.168.77.58 or host 192.168.77.59 or host 192.16
8.77.61 or host 192.168.77.63 or host 192.168.77.65 or host 192.168.77.70 or host 192.168.77.71 or host 192.168.77.74 or host 192.168.77.75 or host 192.168.77.78 or host 192.168.77.84 or host 192.168.77.85 or host 192.168.77.88 or host 192.168.77.89 or host 192.168.77.90
or host 192.168.77.93 or host 192.168.77.96 or host 192.168.70.97 or host 192.168.77.97 or host 192.168.77.99 or host 192.168.77.139 and host 172.17.0.20 and proto ICMP)) &&
        - (not (host 192.168.7.103 or host 192.168.7.105 or host 192.168.7.109 or host 192.168.7.110 or host 192.168.7.112 or host 192.168.7.113 or host 192.168.7.116 or host 192.168.7.117 or host 192.168.7.118 or host 192.168.7.119 or host 192.168.7.156 or host 192.168.7
.173 and host 192.168.7.235 and proto ICMP)) &&
        - (not (host 192.168.7.46 or host 192.168.7.68 or host 192.168.7.69 or host 192.168.7.87 or host 192.168.7.91 or host 192.168.7.92 or host 192.168.7.93 or host 192.168.7.126 or host 192.168.70.20 or host 192.168.70.31 or host 192.168.70.32 or host 192.168.70.33 or
 host 192.168.70.34 or host 192.168.70.35 or host 192.168.70.38 or host 192.168.70.39 or host 192.168.70.43 or host 192.168.70.45 or host 192.168.70.46 or host 192.168.70.50 or host 192.168.70.56 or host 192.168.70.64 or host 192.168.70.65 or host 192.168.70.66 or host 19
2.168.70.67 or host 192.168.70.91 or host 192.168.70.93 or host 192.168.70.95 or host 192.168.70.96 or host 192.168.77.101 or host 192.168.77.102 or host 192.168.77.110 or host 192.168.77.111 or host 192.168.77.113 or host 192.168.77.115 or host 192.168.77.140 and host 17
2.17.0.19 and port 3128)) &&
        - (not (host 192.168.70.35 or host 192.168.70.66 and host 192.168.7.86 and port 139)) &&
        - (not (host 192.168.70.33 or host 192.168.70.35 or host 192.168.70.66 or host 192.168.70.67 and host 192.168.7.86 and port 445)) &&
        - (not (host 192.168.7.95 or host 192.168.7.172 or host 192.168.7.173 and host 192.168.7.90 and port 139)) &&
        - (not (host 192.168.7.90 or host 192.168.7.201 or host 192.168.70.35 and host 192.168.7.94 and port 139)) &&
        - (not (host 192.168.70.66 and host 192.168.7.122 and port 445)) &&       - (not (host 192.168.7.91 or host 192.168.70.33 or host 192.168.70.35 or host 192.168.70.66 and host 192.168.7.136 and port 139)) &&
        - (not (host 192.168.7.68 and host 192.168.7.136 and port 445)) &&
        - (not (host 192.168.70.66 and host 192.168.7.140 and port 139)) &&
        - (not (host 192.168.70.35 or host 192.168.70.66 and host 192.168.7.143 and port 139)) &&
        - (not (host 192.168.7.68 or host 192.168.70.35 or host 192.168.70.66 and host 192.168.7.159 and port 139)) &&
        - (not (host 192.168.70.66 and host 192.168.7.161 and port 139)) &&
        - (not (host 192.168.7.68 or host 192.168.7.92 or host 192.168.70.35 or host 192.168.70.66 and host 192.168.7.175 and port 139)) &&
        - (not (host 192.168.70.66 and host 192.168.7.179 and port 139)) &&
        - (not (host 192.168.70.33 or host 192.168.70.35 or host 192.168.70.66 or host 192.168.70.67 and host 192.168.7.192 and port 139)) &&
        - (not (host 192.168.70.33 and host 192.168.7.192 and port 445)) &&
        - (not (host 192.168.70.66 and host 192.168.7.196 and port 139)) &&
        - (not (host 192.168.70.66 and host 192.168.7.196 and port 445)) &&
        - (not (host 192.168.70.35 or host 192.168.70.66 and host 192.168.7.198 and port 139)) &&
        - (not (host 192.168.70.33 or host 192.168.70.35 or host 192.168.70.66 and host 192.168.7.199 and port 445)) &&
        - (not (host 192.168.70.66 or host 192.168.70.67 and host 192.168.7.199 and port 139)) &&
        - (not (host 192.168.7.95 or host 192.168.7.172 or host 192.168.7.173 and host 192.168.7.201 and port 139)) &&
        - (not (host 192.168.70.33 or host 192.168.70.35 or host 192.168.70.66 and host 192.168.7.204 and port 139)) &&
        - (not (host 192.168.7.199 and host 192.168.7.234 and port 139)) &&
        - (not (host 192.168.7.92 and host 10.255.255.2 and port 3128)) &&
        - (not (host 192.168.7.46 or host 192.168.7.68 or host 192.168.7.69 or host 192.168.7.87 or host 192.168.7.91 or host 192.168.7.92 or host 192.168.7.93 or host 192.168.7.126 or host 192.168.70.20 or host 192.168.70.31 or host 192.168.70.32 or host 192.168.70.33 or
 host 192.168.70.34 or host 192.168.70.35 or host 192.168.70.38 or host 192.168.70.39 or host 192.168.70.43 or host 192.168.70.45 or host 192.168.70.46 or host 192.168.70.50 or host 192.168.70.56 or host 192.168.70.64 or host 192.168.70.65 or host 192.168.70.66 or host 19
2.168.70.67 or host 192.168.70.91 or host 192.168.70.93 or host 192.168.70.95 or host 192.168.70.96 or host 192.168.77.101 or host 192.168.77.102 or host 192.168.77.110 or host 192.168.77.111 or host 192.168.77.113 or host 192.168.77.115 or host 192.168.77.140 and host 17
2.17.0.19 and port 80)) &&
        - not port 161 &&
        - not host 192.168.4.88

And suricata log on sensor:
4/5/2022 -- 06:25:31 - <Info> - Using BPF '(not (host 192.168.11.5 or host 192.168.11.11 or host 192.168.11.88 or host 192.168.11.95 or host 192.168.11.134 or host 192.168.11.209 and proto ICMP)) && (not (host 192.168.7.14 or host 192.168.7.45 or host 192.168.7.46 or host 192.168.7.56 or host 192.168.7.57 or 192.168.7.58 or host 192.168.7.68 or host 192.168.7.69 or host 192.168.7.85 or host 192.168.7.87 or host 192.168.7.91 or host 192.168.7.92 or host 192.168.7.93 or host 192.168.7.125 or host 192.168.7.126 or host 192.168.70.19 or host 192.168.70.20 or host 192.168.70.24 or host 192.168.70.26 or host 192.168.70.29 or host 192.168.70.31 or host 192.168.70.32 or host 192.168.70.33 or host 192.168.70.34 or host 192.168.70.35 or host 192.168.70.38 or host 192.168.70.39 or host 192.168.70.43 or host 192.168.70.45 or host 192.168.70.50 or host 192.168.70.53 or host 192.168.70.55 or host 192.168.70.56 or host 192.168.70.61 or host 192.168.70.63 or host 192.168.70.64 or host 192.168.70.65 or host 192.168.70.66 or host 192.168.70.67 or host 192.168.70.88 or host 192.168.70.91 or host 192.168.70.93 or host 192.168.70.95 or host 192.168.70.96 or host 192.168.70.97 or host 192.168.70.98 or host 192.168.77.55 or host 192.168.77.57 or host 192.168.77.101 or host 192.168.77.102 or host 192.168.77.110 or host 192.168.77.111 or host 192.168.77.113 or host 192.168.77.115 or host 192.168.77.140 and host 172.17.0.3 and port 53 or port 139 or port 445)) && (not (host 192.168.7.14 or host 192.168.7.33 or host 192.168.7.35 or host 192.168.7.45 or host 192.168.7.46 or host 192.168.7.69 or host 192.168.7.87 or host 192.168.7.91 or host 192.168.7.92 or host 192.168.70.19 or host 192.168.70.20 or host 192.168.70.24 or host 192.168.70.35 or host 192.168.70.43 or host 192.168.70.26 or host 192.168.70.32 or host 192.168.70.33 or host 192.168.70.34 or host 192.168.70.35 or host 192.168.70.38 or host 192.168.70.45 or host 192.168.70.64 or host 192.168.70.67 or host 192.168.70.87 or host 192.168.70.96 or host 192.168.77.102 and host 172.17.0.
4/5/2022 -- 06:25:51 - <Error> - [ERRCODE: SC_ERR_AFP_CREATE(190)] - Failed to attach filter: Invalid argument
4/5/2022 -- 06:25:51 - <Error> - [ERRCODE: SC_ERR_AFP_CREATE(190)] - Couldn't init AF_PACKET socket, fatal error
4/5/2022 -- 06:25:51 - <Error> - [ERRCODE: SC_ERR_FATAL(171)] - thread W#01-bond0 failed

I noticed that the error occurs when the bpf file on the sensor is longer than 1090 characters

**Comment by mnasec** [comment]
I have a question about what do you want to archive:

- a) Stop all alarms from this set of filters, not mattering which alarm
- b) Stop some special alarms because they are to noisy

In my case i had some special rules that are very noisy to some ip addresses, so i suppressed the alarms in the global.sls file by src or dst ip like the following:
```
thresholding:
  sids:
    2013028:
      #Meraki curl User-Agent Outbound - Cloud Access
      - suppress:
          gen_id: 1
          track: by_dst
          ip: 209.206.48.0/20
    2013031:
      #Python-urllib/ Suspicious User Agent - update ironports - cisco esa
      - suppress:
          gen_id: 1
          track: by_src
          ip: 10.11.12.13,10.14.15.16
```
This solved the noisy alarm rules for me without using a "big" bfp filter.

I am only using bpf for keeping space with a steno filter and shutting off noisy snmp ;-)
```
steno:
  bpf:
    - not port 443
nids:
  bpf:
    - not udp port 161
```

Documented in https://docs.securityonion.net/en/2.3/managing-alerts.html?highlight=threshold#threshold

Possible this may help you
